floodlight/io/sportradar.py

```diff
@@ -0,0 +1,199 @@
+    tID_link = {}
+    home_away_link = {}
+    for competitor in events["statistics"]["totals"]["competitors"]:
+        tID_link.update({competitor["id"]: competitor["qualifier"]})
+        home_away_link.update(
+            {competitor["qualifier"]: (competitor["id"], competitor["name"])}
+        )
+
+    # extract periods
+    periods = set(
+        [event["period_name"] for event in timeline if event["type"] == "period_start"]
+    )
+
+    # create team event dict
+    columns = [
+        "eID",
+        "gameclock",
+        "time_stamp",
+        "minute",
+        "second",
+        "pID",
+        "player_name",
+        "tID",
+        "team_name",
+        "mID",
+        "event_name",
+        "home_score",
+        "away_score",
+        "scorer",
+        "assists",
+        "zone",
+        "shot_type",
+        "outcome",
+        "players",
+    ]
+
+    # seems unnecessary
+    segments = [f"HT{period}" for period in periods]
+
+    team_event_lists = {
+        team: {segment: {col: [] for col in columns} for segment in segments}
+        for team in teams
+    }
+
+    period = None
+    # loop
+    for event in timeline:
+        if not period:
+            # get first period
+            if event["type"] == "period_start":
+                period = event["period_name"]
+                segment = f"HT{period}"
+                segment_start = datetime.datetime.fromisoformat(event["time"])
+            else:
+                # skip events before first period starts
+                continue
+        # get new periods
+        else:
+            if event["type"] == "period_start":
+                period = event["period_name"]
+                segment = f"HT{period}"
+                segment_start = datetime.datetime.fromisoformat(event["time"])
+
+        # extract event, player and team ids and names
+        eID = event["id"]
+
+        # add all teams as competitors if no competitor is specified in event
+        competitor = [event["competitor"]] if "competitor" in event else teams
+
+        tID = home_away_link[competitor[0]][0] if len(competitor) == 1 else None
+        team_name = home_away_link[competitor[0]][1] if len(competitor) == 1 else None
+        pID = event["player"]["id"] if "player" in event else None
+        player_name = event["player"]["name"] if pID else None
+
+        event_name = event["type"]
+
+        # extract time codes and matchclock
+        time_stamp = datetime.datetime.fromisoformat(event["time"])
+        time_delta = time_stamp - segment_start
+        gameclock = time_delta.seconds
+        if "match_clock" in event:
+            match_clock = event["match_clock"]
+            minute, second = [int(x) for x in match_clock.split(":")]
+        else:
+            minute, second = (None, None)
+
+        # extract optional event information
+        outcome = event["outcome"] if "outcome" in event else None
+        home_score = event["home_score"] if "home_score" in event else None
```

Comment:
for all commands with this logic: there's the dict' ```.get()``` method you can use to simplify this, i.e. ```event.get("home_score")```. also, you may use the ```get_and_convert()``` function from io/utils.py who does the same including a type conversion. that way, you could store "home_score" directly as int instead of float

Reply:
Thanks thats great! I implemented `get_and_convert()` for the non-nested dicts. However, it seems like pandas converts them back to float whenever the also is an NaN in the same column

Reply:
Yes, pandas is a bit unsatisfactory (imho) concerning data types, Makes sense though as types need to be consistent across columns and np.nan is treated as float.